### PR TITLE
add editorial for atcoder dp contest + small wording fixes

### PR DESCRIPTION
**Pull request by nwatx**
Like [last time](https://github.com/cpinitiative/usaco-guide/pull/998), I guess screenshot the blog directly to review it. A link can be found [here](https://nwatx.me/post/atcoderdp), or you can PR [source](https://github.com/nwatx/nwatx-blog/blob/main/content/atcoderdp.mdx) directly I guess.

### CODE REVIEW

**Review by SansPapyrus683 — APPROVED** [review]

File: content/4_Gold/Intro_DP.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 id: intro-dp
 title: 'Introduction to DP'
-author: Michael Cao, Benjamin Qi
+author: Michael Cao, Benjamin Qi, Neo Wang
```
Comment (nwatx) [inline content/4_Gold/Intro_DP.mdx:4]:
```suggestion
author: Michael Cao, Benjamin Qi
```

**Review by dongliu0426 — APPROVED** [review]

**Comment by jessechoe10** [comment]
Can I add my solutions for these problems as well?

**Comment by jessechoe10** [comment]
> Can I add my solutions for these problems as well?

I have solutions for F - I.

**Comment by nwatx** [comment]
> Can I add my solutions for these problems as well?

submit a pr to https://github.com/nwatx/nwatx-blog/blob/main/content/atcoderdp.mdx

**Comment by jessechoe10** [comment]
> > Can I add my solutions for these problems as well?
> 
> @jessechoe10 submit a pr to https://github.com/nwatx/nwatx-blog/blob/main/content/atcoderdp.mdx

Whats wrong with linking to my github?

**Review by HarryWangATX — APPROVED** [review]

**Review by jessechoe10 — APPROVED** [review]

File: content/4_Gold/Intro_DP.mdx [hunk 2]
```diff
@@ -128,16 +125,15 @@ problemsetters rarely set direct applications of them.
 		url="https://atcoder.jp/contests/dp/tasks"
 		starred
 	>
-		Some tasks are beyond the scope of Gold.
+		Some tasks are beyond the scope of Gold. An editorial for problems A-I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A - L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).
```
Comment (dolphingarlic) [inline content/4_Gold/Intro_DP.mdx:128]:
```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A to L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).
```

File: content/4_Gold/Intro_DP.mdx [hunk 3]
```diff
@@ -97,7 +94,7 @@ against.
 
 ## Classical Problems
 
-The next few modules provide examples of some classical problems, or Dynamic
+The next few modules provide examples of some classical problems: Dynamic
```
Comment (dolphingarlic) [inline content/4_Gold/Intro_DP.mdx:97]: A colon doesn't really make sense here. Maybe use brackets instead?

File: content/4_Gold/Intro_DP.mdx [hunk 4]
```diff
@@ -58,34 +55,34 @@ If you prefer watching videos instead, here are some options:
...
 	>
-		more DP videos, but some are more related to interview questions than CP
+		Animated DP videos that pertain to interview questions
 	</Resource>
 </Resources>
 
 <Info title="Pro Tip">
 
-It's usually a good idea to write a slower solution first. For example, if the
+Usually, it is a good idea to write a slower solution first. For example, if the
```
Comment (dolphingarlic) [inline content/4_Gold/Intro_DP.mdx:85]: I prefer the original

**Review by dolphingarlic — CHANGES_REQUESTED** [review]

File: content/4_Gold/Intro_DP.mdx [hunk 5]
```diff
@@ -128,16 +125,15 @@ problemsetters rarely set direct applications of them.
 		url="https://atcoder.jp/contests/dp/tasks"
 		starred
 	>
-		Some tasks are beyond the scope of Gold.
+		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A to L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).
```
Comment (nwatx) [inline content/4_Gold/Intro_DP.mdx:128]: @jessechoe10 doesn't contain J

Comment (jessechoe10) [reply content/4_Gold/Intro_DP.mdx:128]: sheesh i can add later.

Comment (jessechoe10) [reply content/4_Gold/Intro_DP.mdx:128]: I'll add more solutions later lol (it shouldn't be too hard to add them)

Comment (HarryWangATX) [reply content/4_Gold/Intro_DP.mdx:128]: PRs should reflect the current state 👀

Comment (jessechoe10) [reply content/4_Gold/Intro_DP.mdx:128]: What should I change it to then?

Comment (HarryWangATX) [reply content/4_Gold/Intro_DP.mdx:128]: nwatx has A-O

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
@nwatx link problem L, turn on C++ for problem M

**Comment by nwatx** [comment]
> @nwatx link problem L, turn on C++ for problem M

deploying... give ~40 seconds

**Review by HarryWangATX — APPROVED** [review]
looks good

**Comment by maggie-j-liu** [comment]
> This post contains the editorials for tasks A-E contained in the AtCoder DP Contest, since there is no official editorial.

*https://nwatx.me/post/atcoderdp*

change to A-O

File: content/4_Gold/Intro_DP.mdx [hunk 6]
```diff
@@ -128,16 +125,15 @@ problemsetters rarely set direct applications of them.
 		url="https://atcoder.jp/contests/dp/tasks"
 		starred
 	>
-		Some tasks are beyond the scope of Gold.
+		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp).
```
Comment (HarryWangATX) [inline content/4_Gold/Intro_DP.mdx:128]:
```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to O can be found [here](https://nwatx.me/post/atcoderdp).
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/4_Gold/Intro_DP.mdx (hunk 3 above)
Comment (nwatx) [reply content/4_Gold/Intro_DP.mdx:97]: why not?

File: content/4_Gold/Intro_DP.mdx (hunk 6 above)
Comment (jessechoe10) [reply content/4_Gold/Intro_DP.mdx]: Why?

File: content/4_Gold/Intro_DP.mdx [hunk 7]
```diff
@@ -128,16 +125,15 @@ problemsetters rarely set direct applications of them.
 		url="https://atcoder.jp/contests/dp/tasks"
 		starred
 	>
-		Some tasks are beyond the scope of Gold.
+		Some tasks are beyond the scope of Gold. Editorials for problems A to O can be found [here](https://nwatx.me/post/atcoderdp).
```
Comment (maggie-j-liu) [inline content/4_Gold/Intro_DP.mdx:128]:
```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to U can be found [here](https://nwatx.me/post/atcoderdp).
```

Comment (dongliu0426) [reply content/4_Gold/Intro_DP.mdx]: lol

**Comment by maggie-j-liu** [comment]
code formatting is not consistent, for example 
image

can you add spaces around all operators?

**Comment by nwatx** [comment]
> code formatting is not consistent, for example
> image
> 
> can you add spaces around all operators?

done

File: content/4_Gold/Intro_DP.mdx (hunk 5 above)
Comment (sathu-beep) [reply content/4_Gold/Intro_DP.mdx]: sussy irl

**Review by sathu-beep — APPROVED** [review]
ez gang rn

**Comment by maggie-j-liu** [comment]
missed one (this is in j)
image

also, some of them don't have time complexity (most of the ones after j)

**Comment by nwatx** [comment]
> missed one (this is in j)
> image
> 
> also, some of them don't have time complexity (most of the ones after j)

done

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**